pep-9999.txt

```diff
@@ -0,0 +1,171 @@
+PEP: 9999
+Title: Python local packages directory [We need to find a proper name]
+Version: $Revision$
+Last-Modified: $Date$
+Author: Kushal Das <mail@kushaldas.in>, Steve Dower <steve.dower@python.org>,
+        Donald Stufft <donald@stufft.io>, Nick Coghlan <ncoghlan@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 16-May-2018
+Python-Version: 3.8
+
+
+
+Abstract
+========
+
+This PEP proposes to add to Python a mechanism to have a ``__pypackages__``
+directory  per project to have its own independent set of installed packages.
+This will share the standard library with the base installed Python. This will
+avoid the steps to create or activate or deactivate "virtual environments". Any
+Python scripts will use the ``__pypackages__`` from the base directory of the 
+script.
+
+
+
+Motivation
+==========
+
+Python virtual environments have become an essential part of development and
+teaching workflow in the community, but at the same time, they create a
+barrier to entry for many folks. The following are few examples of the issues
+people get into while being introduced to Python (or programming for the first
+time).
+
+- How virtual environments work is a lot of information for anyone new. It takes
+  a lot of extra time and effort to explain them.
+
+- Different platforms and shell environments require different sets of commands
+  to work on the virtual environments. Any workshop or teaching environment with
+  people coming with different operating systems installed on the laptops create a
+  lot of confusion among the participants.
```

Comment:
Anyone have any good references for this? Would love to link to some instructions from a workshop about this - I think it'd help make the point.

Reply:
I don't think the PEP really solves this particular problem – there's still `python`, `py` or `python3`. Virtualenvs require awkward OS-specific commands to create and activate, but once activated they work the same everywhere.

Reply:
@zooba will linking to https://virtualenv.pypa.io/en/stable/userguide/#activate-script do?